doc/development/python-release.md

```diff
@@ -0,0 +1,67 @@
+# Python Release Process
```

Comment:
Maybe rename the file `python-release-process.md` just to be 100% clear.

Reply:
done